internal/config/config_test.go

```diff
@@ -0,0 +1,253 @@
+				Expect(cfgErr).To(MatchError(`plugin requires minimum API version v3.99, got: "3.98.0"`))
+			})
+		})
+
+		When("API version is too high", func() {
+			BeforeEach(func() {
+				fakeCLIConnection.ApiVersionReturns("4", nil)
+			})
+
+			It("returns an error", func() {
+				Expect(cfgErr).To(MatchError(`plugin requires API major version v3, got: "4.0.0"`))
+			})
+		})
+
+		When("version is not parsable", func() {
+			BeforeEach(func() {
+				fakeCLIConnection.ApiVersionReturns("zenobia", nil)
+			})
+
+			It("returns an error", func() {
+				Expect(cfgErr).To(MatchError("error parsing API version: Malformed version: zenobia"))
+			})
+		})
+
+		When("error getting version", func() {
+			BeforeEach(func() {
+				fakeCLIConnection.ApiVersionReturns("", fmt.Errorf("boom"))
+			})
+
+			It("returns an error", func() {
+				Expect(cfgErr).To(MatchError("error retrieving API version: boom"))
+			})
+		})
+	})
+
+	Describe("access token", func() {
+		BeforeEach(func() {
+			fakeCLIConnection.AccessTokenReturns("fake-token", nil)
+		})
+
+		It("gets the access token", func() {
+			Expect(cfg.APIToken).To(Equal("fake-token"))
+			Expect(cfgErr).NotTo(HaveOccurred())
+		})
+
+		When("error getting access token", func() {
+			BeforeEach(func() {
+				fakeCLIConnection.AccessTokenReturns("", fmt.Errorf("boom"))
+			})
+
+			It("returns the error", func() {
+				Expect(cfgErr).To(MatchError("error reading access token: boom"))
+				Expect(cfg.APIToken).To(Equal(""))
+			})
+		})
+	})
+
+	Describe("API endpoint", func() {
+		BeforeEach(func() {
+			fakeCLIConnection.ApiEndpointReturns("fake-api-endpoint", nil)
+		})
+
+		It("gets the API endpoint", func() {
+			Expect(cfg.APIEndpoint).To(Equal("fake-api-endpoint"))
+			Expect(cfgErr).NotTo(HaveOccurred())
+		})
+
+		When("error getting API endpoint", func() {
+			BeforeEach(func() {
+				fakeCLIConnection.ApiEndpointReturns("", fmt.Errorf("boom"))
+			})
+
+			It("returns the error", func() {
+				Expect(cfgErr).To(MatchError("error reading API endpoint: boom"))
+				Expect(cfg.APIToken).To(Equal(""))
+			})
+		})
+	})
+
+	Describe("skip SSL validation", func() {
+		BeforeEach(func() {
+			fakeCLIConnection.IsSSLDisabledReturns(true, nil)
+		})
+
+		It("gets the skip SSL validation", func() {
+			Expect(cfg.SkipSSLValidation).To(Equal(true))
+			Expect(cfgErr).NotTo(HaveOccurred())
+		})
+
+		When("error getting skip SSL validation", func() {
+			BeforeEach(func() {
+				fakeCLIConnection.IsSSLDisabledReturns(false, fmt.Errorf("boom"))
+			})
+
+			It("returns the error", func() {
+				Expect(cfgErr).To(MatchError("error reading skip SSL validation: boom"))
+				Expect(cfg.SkipSSLValidation).To(Equal(false))
+			})
+		})
+	})
+
+	Describe("parallel upgrades", func() {
+		When("not specified", func() {
+			It("defaults", func() {
+				Expect(cfg.ParallelUpgrades).To(Equal(10))
+			})
+		})
+
+		When("not a number", func() {
+			BeforeEach(func() {
+				fakeArgs = []string{"-parallel", "boudica"}
```

Comment:
I was expecting to see `brokerName` as the first arg here?

Reply:
Fair point, but the test passes, so I guess it hits this error first...